src/arc_utilities/ros_helpers.py

```diff
@@ -39,56 +39,72 @@ def get(self, block_until_data=True):
         Returns the latest msg from the subscribed topic
 
         Parameters:
-            block_until_data (bool): block if no message has been received yet. 
+            block_until_data (bool): block if no message has been received yet.
                                      Guarantees a msg is returned (not None)
         """
         wait_for(lambda: not (block_until_data and self.data is None))
-            
+
         with self.lock:
             return deepcopy(self.data)
 
-    
+
 def wait_for(func):
     """
     Waits for function evaluation to be true. Exits cleanly from ros.
 
     Introduces sleep delay, not recommended for time critical operations
     """
-    
+
     while not func() and not rospy.is_shutdown():
         time.sleep(0.01)
 
-        
-def joy_to_xbox(joy):
+
+def joy_to_xbox(joy, xpad=True):
     """
     Transforms a joystick sensor_msg to a XBox controller for easier code readability
-    
+
     Parameters:
     joy (sensor_msgs/Joy): xbox msg
+    xpad (bool): True if using the default xpad driver, False if using xboxdrv
 
     Returns:
     xbox struct where fields are the button names
     """
     class Xbox_msg():
-        pass
+        def __str__(self):
+            items = vars(self).items()
+            return "\n".join("%s: %s" % item for item in items)
     x = Xbox_msg()
-    x.A, x.B, x.X, x.Y, x.LB, x.RB, \
-        x.back, x.start, x.power,\
-        x.stick_button_left, x.stick_button_right, \
-        x.DL, x.DR, x.DU, x.DD = joy.buttons
-    x.LH, x.LV, x.LT, x.RH, x.RV, x.RT, x.DH, x.DV = joy.axes
+    if xpad:
+        x.A, x.B, x.X, x.Y, x.LB, x.RB, \
+            x.back, x.start, x.power,\
+            x.stick_button_left, x.stick_button_right, \
+            x.DL, x.DR, x.DU, x.DD = joy.buttons
+        x.LH, x.LV, x.RH, x.RV, x.RT, x.LT, x.DH, x.DV = joy.axes
+    else:
+        x.A, x.B, x.X, x.Y, x.LB, x.RB, \
+            x.back, x.start, x.power,\
+            x.stick_button_left, x.stick_button_right = joy.buttons
+        x.LH, x.LV, x.RH, x.RV, x.RT, x.LT, x.DH, x.DV = joy.axes
     return x
 
 
 class Xbox():
-    def __init__(self, joystick_topic="joy"):
+    def __init__(self, joystick_topic="joy", xpad=True):
+        """
+        Parameters:
+        joystick_topic (string): topic name to subscribe to
+        xpad (bool): True if using the default xpad driver, False if using xboxdrv
```

Comment:
I don't remember if realtime is using xpad or xboxdrv, even though you just told me.
I certainly wont remember in the future. 

How would one look this up? Put that in doc string? Or link to the notes you are writing?

Reply:
It's in the Victor code, and it'll be documented.